server/api/diagnose_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/pd/server"
+)
+
+var _ = Suite(&testDiagnoseAPISuite{})
+
+type testDiagnoseAPISuite struct {
+	hc *http.Client
+}
+
+func (s *testDiagnoseAPISuite) SetUpSuite(c *C) {
+	s.hc = newHTTPClient()
+}
+
+func checkDiagnoseResponse(c *C, body []byte) {
+	got := []Recommended{}
+	json.Unmarshal(body, &got)
+	for _, r := range got {
```

Comment:
It seems the response will be empty for a cluster with no abnormal condition, so the test will not check anything.

Reply:
`follow.Close()` ?